x/mongo/driver/topology/connection_test.go

```diff
@@ -110,6 +110,19 @@ func TestConnection(t *testing.T) {
 				assert.NotNil(t, err, "expected connect error %v, got nil", want)
 				assert.Equal(t, want, got, "expected error %v, got %v", want, got)
 			})
+			t.Run("cancelConnectContext is nil after connect", func(t *testing.T) {
+				conn, got := newConnection(context.Background(), address.Address(""))
```

Comment:
We generally use `got` if we want to actually check the error against something we're expecting, but that's not the case here, so let's change this to `con, err := ...` for clarity.